src/PrestaShopBundle/Translation/TranslationCatalogueBuilder.php

```diff
@@ -0,0 +1,317 @@
+        $userTranslatedCatalogue = $provider->getUserTranslatedCatalogue($locale);
+
+        $translations = [];
+
+        foreach ($defaultCatalogueMessages as $domain => $messages) {
+            $missingTranslations = 0;
+            $translations[$domain] = [];
+
+            foreach ($messages as $translationKey => $translationValue) {
+                $data = [
+                    'default' => $translationKey,
+                    'xliff' => $fileTranslatedCatalogue->defines($translationKey, $domain)
+                        ? $fileTranslatedCatalogue->get($translationKey, $domain)
+                        : null,
+                    'database' => $userTranslatedCatalogue->defines($translationKey, $domain)
+                        ? $userTranslatedCatalogue->get($translationKey, $domain)
+                        : null,
+                ];
+
+                // if search is empty or is in catalog default|xliff|database
+                if (empty($search) || $this->dataContainsSearchWord($data, $search)) {
+                    $translations[$domain][$translationKey] = $data;
+
+                    if (null === $data['xliff'] && null === $data['database']) {
+                        ++$missingTranslations;
+                    }
+                }
+            }
+
+            $translations[$domain]['__metadata'] = [
+                'count' => count($translations[$domain]),
+                'missing_translations' => $missingTranslations,
+            ];
+        }
+
+        ksort($translations);
+
+        return $translations;
+    }
+
+    /**
+     * @param string $type
+     * @param string $locale
+     * @param array $search
+     * @param string|null $theme
+     * @param string|null $module
+     *
+     * @return array
+     *
+     * @throws Exception
+     */
+    public function getTree(
+        string $type,
+        string $locale,
+        array $search,
+        ?string $theme,
+        ?string $module
+    ): array {
+        $catalogue = $this->getCatalogue($type, $locale, $search, $theme, $module);
+
+        return $this->translationsTreeBuilder->buildDomainTreeForApi(
+            $catalogue,
+            $locale,
+            $search,
+            $theme,
+            $module
+        );
+    }
+
+    /**
+     * @param string $type
+     * @param string $locale
+     * @param array $search
+     * @param string|null $theme
+     * @param string|null $module
+     *
+     * @throws Exception
+     */
+    private function validateParameters(
+        string $type,
+        string $locale,
+        array $search,
+        ?string $theme,
+        ?string $module
+    ): void {
+        if (!in_array($type, self::ALLOWED_TYPES)) {
+            throw new UnexpectedTranslationTypeException('This \'type\' param is not valid.');
+        }
+        if (self::TYPE_MODULES === $type && empty($module)) {
+            throw new Exception('This \'selected\' param is not valid. Module must be given.');
```

Comment:
```suggestion
            throw new InvalidArgumentException('This \'selected\' param is not valid. Module must be given.');
```
I think this is the right usecase for a InvalidArgumentException 🤔  ?